Firestore/Source/Local/FSTLevelDBRemoteDocumentCache.mm

```diff
@@ -84,6 +84,26 @@ - (nullable FSTMaybeDocument *)entryForKey:(const DocumentKey &)documentKey {
   }
 }
 
+- (MaybeDocumentMap)entriesForKeys:(const DocumentKeySet &)keys {
+  MaybeDocumentMap results;
+
+  LevelDbRemoteDocumentKey currentKey;
+  auto it = _db.currentTransaction->NewIterator();
+
+  for (const DocumentKey &key : keys) {
+    if (key > currentKey.document_key()) {
+      it->Seek([self remoteDocumentKey:key]);
```

Comment:
IIUC, leveldb iterators will seek in any direction? But you're attempting to seek forward only, right? Or at the very least, you're attempting to optimize by eliminating seeks to "entry2" through "entry8"  if the seek to "entry1" ended up putting the iterator at "entry9"?

One caution: This assumes that our idea of ordering is the same as leveldb's. We compare via vector<string>::op<, while leveldb (presumably) uses flattened string comparison. This seems like it should mostly work out, but I'm uncertain as to the edge cases.

Unless you suspect that this optimization will save a lot of time in practice, I'd be tempted to just eliminate it and unconditionally seek. If you suspect that this *is* a significant optimization, then we should probably put some more due diligence in (preferably in the form of tests) that ensures that this ordering is consistent on both sides.

Reply:
Thanks for pointing this out. Actually, it looks like `Seek` always searches from the beginning. I agree that this optimization likely is not worth it. Removed.